agv_robot_joint_publisher:
  type: "joint_state_controller/JointStateController"
  publish_rate: 50

agv_ackermann_steering_controller:
  type: "agv_ackermann_steering_controller/AgvAckermannSteeringController"
  rear_left_wheel: 'left_wheel_joint'
  rear_right_wheel: 'right_wheel_joint'
  publish_rate: 50
  pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
  front_steer: 'front_steering_joint'
  front_wheel: 'front_wheel_joint'
  # Wheel separation and diameter. These are both optional.
  # diff_drive_controller will attempt to read either one or both from the
  # URDF if not specified as a parameter
  wheel_separation_h : 1.25 # meters 
  wheel_radius : 0.125    # meters
  
   # Wheel separation and radius multipliers
  wheel_separation_h_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0

  # steer pos multiplier
  steer_pos_multiplier: 1.0
  
  # Velocity commands timeout [s], default 0.5
  cmd_vel_timeout: 0.5
  velocity_rolling_window_size: 10

  # 
  allow_multiple_cmd_vel_publishers: true

  # Base frame_id
  base_frame_id: base_footprint

  #Publish to TF directly or not
  enable_odom_tf: true

  #Name of frame to publish odometry in
  odom_frame_id: odom

  # Husky hardware provides wheel velocities
  estimate_velocity_from_position: false

  # Publish the velocity command to be executed. 
  # It is to monitor the effect of limiters on the controller input.
  publish_cmd: true

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 3.0 #1.0  # m/s
      min_velocity           : -3.0 #-0.5 # m/s
      has_acceleration_limits: true
      max_acceleration       : 3 #0.8  # m/s^2
      min_acceleration       : -3 #-0.4 # m/s^2
      has_jerk_limits        : false
      max_jerk               : 5.0  # m/s^3
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 3.0 #1.7  # rad/s
      has_acceleration_limits: true
      max_acceleration       : 6.0  # rad/s^2
      has_jerk_limits        : false
      max_jerk               : 2.5  # rad/s^3

gazebo_ros_control/pid_gains:
  disk_link_joint:
    p: 2.0
    i: 0.0
    d: 0.5
  steer_wheel_joint:
    p: 0.1
    i: 0.0
    d: 0.0

# Publish all joint states -----------------------------------
#joint_state_controller:
#  type: joint_state_controller/JointStateController
#  publish_rate: 50  

# Velocity Controllers ---------------------------------------
#steer_velocity_controller:
#  type: velocity_controllers/JointVelocityController
#  joint: steer_wheel_joint
#  pid: {p: 5.0, i: 0.0, d: 0.0}
  
# Velocity Controllers ---------------------------------------
#steer_direction_controller:
#  type: position_controllers/JointPositionController
#  joint: disk_link_joint
#  pid: {p: 10.0, i: 0.01, d: 1.0}
